spec/consensus/gossip/files.md

```diff
@@ -0,0 +1,125 @@
+# Files and Types
+
+Files and types related to the State/Gossip interaction, excluding testing related.
+
+## reactor.go
+- Entry point for the consensus reactor (State+Gossip)
+    - `NewReactor`
+    - `OnStart`
+    - `OnStop`
+    - `SwitchToConsensus`
+
+- Implements the Gossip Layer
+    - Peer management
+        - `InitPeer`
+        - `AddPeer`
+        - `RemovePeer`
+
+    - Message delivery
+        - `Receive`
+            - Processes messages delivered by the Gossip layer and updates consensus (`State`) and peer information (`PeerState`)
+            - Messages are received through multiple channels
+                - `StateChannel`: `NewRoundStepMessage`, `NewValidBlockMessage`, `HasVoteMessage`, `VoteSetMaj23Message`
+                - `DataChannel`: `ProposalMessage`, `ProposalPOLMessage`, `BlockPartMessage`
+                - `VoteChannel`: `VoteMessage`
+                - `VoteBitsChannel`: `VoteSetBitsMessage`
+
+        - `PeerState` update
+            - `ApplyNewRoundStepMessage`
+            - `ApplyNewValidBlockMessage`
+            - `ApplyHasVoteMessage`
+            - `SetHasProposal`
+            - `ApplyProposalPOLMessage`
+            - `SetHasProposalBlockParts`
+            - `SetHasVote`
+            - `ApplyVoteSetBitsMessage`
+
+        - Peer information requests
+            - `VoteSetMaj23Message` (CODE TODO: Move handling from receive to helper function.)
+
+    - [Consensus state](#state.go) update
+
+    - Consensus state monitoring
+        - `subscribeToBroadcastEvents` - event bus (Why not simply using channels and go-routines?)
+            - `broadcastNewRoundStepMessage`
+            - `broadcastNewValidBlockMessage`
+            - `broadcastHasVoteMessage`
+        - `go updateRoundStateRoutine`
+
+
+    - [Consensus state](#state.go) gossip 
+        - `conR.gossipDataRoutine`
+            - `BlockPart`
+                 - if peers agree on block parts hash
+                 - if peer is lagging, through `gossipDataForCatchup`
+            - `Proposal`
+            - `ProposalPOL`
+	    - `conR.gossipVotesRoutine`
+            - `PickSendVote` - sends a `Vote` of the specified set, which depends on the step of self and peer
+                - `gossipvotesForHeight` - if H matches
+                    - `LastCommit`
+                    - POL `Prevotes`
+                    - `Prevotes`
+                    - `Precommits`
+
+	    - `conR.queryMaj23Routine`
```

Comment:
Intended to handle Byzantine behavior, more specifically, the situation on which validators produce votes for different values in the same height, round, and step.